content/en/platform/corda/5.0-dev-preview-2/developing/api/api-persistence.md

```diff
@@ -8,3 +8,173 @@ menu:
+
+  The migrations run when the virtual node is created and logging shows the migrations executing.
+  If you have direct database access, you should see the tables being created.
+  If you are using Postgres, make sure to look under the correct schema, since each virtual node creates a new schema, unless an external VNode DB was provided during VNode creation.
+
+## Mapping your new tables to JPA entities
+
+CorDapps should use JPA annotated POJOs for data access objects.
+Each class requires `@CordaSerializable` and `@Entity` annotations.
+The following is an example that also defines some named JPQL queries:
+
+```java
+@CordaSerializable
+@Entity(name = "dog")
+@NamedQueries(
+    NamedQuery(name = "Dog.summon", query = "SELECT d FROM Dog d WHERE d.name = :name"),    
+    NamedQuery(name = "Dog.independent", query = "SELECT d FROM Dog d WHERE d.owner IS NULL"),    
+    NamedQuery(name = "Dog.summonLike", query = "SELECT d FROM Dog d WHERE d.name LIKE :name ORDER BY d.name"),    
+    NamedQuery(name = "Dog.all", query = "SELECT d FROM Dog d ORDER BY d.name"),    
+    NamedQuery(name = "Dog.release", query = "UPDATE Dog SET owner=null")
+)
+class Dog(
+    @Id    
+    @Column(name = "id", nullable = false)    
+    val id: UUID,    
+    @Column(name = "name")    
+    val name: String,    
+    @Column(name = "birthdate")    
+    val birthdate: Instant,
+    @Column(name = "owner")
+    val owner: String?
+) {
+    constructor() : this(id = UUID.randomUUID(), name = "", birthdate = Instant.now(), owner = "")
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Dog
+
+        if (id != other.id) return false
+        if (name != other.name) return false
+        if (birthdate != other.birthdate) return false
+        if (owner != other.owner) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = id.hashCode()
+        result = 31 * result + name.hashCode()
+        result = 31 * result + birthdate.hashCode()
+        result = 31 * result + owner.hashCode()
+        return result
+    }
+}
+```
+
+{{< note >}}
+This example specifies the table and column names.
+If they are not specified, JPA uses defaults.
+{{< /note >}}
+
+
+## Using the persistence API from a CorDapp flow
+
+1. Define a reference to the persistence service. This should be supplied via the Corda dependency injection system:
+  ```java
+  import net.corda.v5.application.flows.CordaInject
+  import net.corda.v5.application.flows.RPCStartableFlow
+  import net.corda.v5.application.persistence.PersistenceService
+
+  class MyExampleFlow : RPCStartableFlow {
+      @CordaInject
+      lateinit var persistenceService: PersistenceService
+
+      // your code goes here
+  ```
+
+2. To create a Dog entity that writes a row to the database:
```

Comment:
You capitalise "Dog" here but you're not consistent in other places. Please change everywhere to either "Dog" or "dog".